### DISCUSSION TITLE
How would I protect a route from being show to the user if they aren't authenticated?

### DISCUSSION THREAD

**Discussion by jaooe** [vercel/next.js#15125] [Help]
For example, 
I have an admin dashboard that I don't want to be shown to unauthenticated users, or even users who aren't an admin... 
How would I stop this route/page from being generated and shown.

**Comment by abhik-b** [comment]
I also have the same question but still unable to find answer.
So my suggestion is that **unless you find a solution to this problem, try this**:

- have a context for whether user is authenticated or not and if  authenticated then whether he is admin or not 
- then get that context in the dashboard page (the page you don't wanna show to non-admins) and if the context returns user as non admin or null then router push to login page or wherever you wanna send 

YOU CAN EVEN CREATE A HOC AND THEN WRAP THOSE PAGES YOU DONT WANNA SHOW TO ALL.

 [_HERE IS A ARTICLE BY SOMEONE WHICH MIGHT SOLVE THIS PROBLEM_ ,PLEASE READ IT ](https://dev.to/justincy/client-side-and-server-side-redirection-in-next-js-3ile)

**Reply by jaooe** [reply]
Thanks, that's exactly what I was looking for.

**Reply by jaooe** [reply]
I'm not sure how static generation affects this though, if I redirect using `getInitialProps` does that stop static generation from rendering my component?
